Expensify/App
# Add password to AddDebitCardPage in NewDot

### Details
Adds a password requirement on the AddDebitCardPage.
We are ignoring the forgot password flow for now, as we still haven't landed on a solution.

### Fixed Issues
https://github.com/Expensify/Expensify/issues/188051 (partial)

### Tests
1. Pull the changes in [Auth](https://github.com/Expensify/Auth/pull/6271) and [Web-Secure](https://github.com/Expensify/Web-Secure/pull/2124)
2. Log into NewDot and navigate to `Settings > Payments > Add payment method > Debit card`
2. Try to submit an empty form and verify that you see an error for the password input.
3. Now fill in all the necessary information (enter 4242 4242 4242 4242 for credit card number), but use a wrong password.
4. Submit the form and verify that you see the following message: `Incorrect Expensify password entered`.
5. Now enter your correct Expensify password and submit the form again.
6. Verify that the card was added.

### QA Steps
Please note that actual password enforcement will only happen after these two PRs are merged as well: https://github.com/Expensify/Auth/pull/6271 and https://github.com/Expensify/Web-Secure/pull/2124. Before that, entering a wrong password wouldn't display an error and the card will still be added.

Follow steps above.

### Tested On


### Screenshots


#### Web

https://user-images.githubusercontent.com/22219519/148591820-5ca7b892-a9bc-413c-b4b7-f7e32a8cacf1.mov

#### Mobile Web


#### Desktop

https://user-images.githubusercontent.com/22219519/148591833-02f90fd7-f3d0-49d3-8623-a01e88421b1c.mov

#### iOS

https://user-images.githubusercontent.com/22219519/148593278-19b218aa-1d7b-4193-ae11-fa9a08b4b5f1.mp4

#### Android

https://user-images.githubusercontent.com/22219519/148596836-7155b605-69b5-454a-ae62-1ccdd55a45be.mov
